docs/developer/architecture/core/uisettings-service.asciidoc

```diff
@@ -3,14 +3,71 @@
 
 NOTE: The UI settings service is available both server and client side.
 
-=== Server side usage
+=== Overview
+
+UI settings are configurable from the Advanced Settings page in Management and control the behavior of {kib}. uiSettings are stored in a config saved object and, as such, conform to the same conditions as other <<saved-objects-service, Saved Objects>>.
+
+There are several ways to configure an advanced setting:
+
+- <<advanced-settings-ui, Through the Advanced Settings UI>>
+- <<uisettings-overrides, Locked via kibana.yml's uiSettings.overrides>>
+- <<client-side-usage, Through the client-side uiSettings Service>>
+- <<server-side-usage, Through the server-side uiSettings Service>>
+
+Keep in mind that once you add a new advanced setting, you cannot change or remove it without <<uisettings-migrations, registering a migration in core>>.
+
+[[advanced-settings-ui]]
+=== Configuration with Advanced Settings UI
+The uiSettings service is the programmatic interface to Kibana's Advanced Settings UI. Kibana plugins use the service to extend Kibana UI Settings Management with custom settings for their plugin.
+
+Configuration through the Advanced Settings UI is restricted to users authorised to acces the Advanced Settings page. Users who don't have permissions to change these values default to using the settings configured to the space they are in. Config saved objects can be shared between spaces.
+
+[[uisettings-overrides]]
+=== Configuration with UI settings overrides
```

Comment:
I was in two minds about explicitly mentioning overrides. We hadn't documented these in the past on purpose to limit their use. Folks generally discovered these by going through the code and if they had and something went wrong, we have to dig through and figure out where the issue is.

Explicitly documenting overrides here gives us the opportunity to warn about misconfiguration issues, limited validation, and how setting an override affects Kibana's behavior.

Reply:
Maybe we add an `experimental[]` flag to this as a middle ground?